react-native-web@^0.13.0 peer dependency breaks npm install for Expo 43

# Bug report

<!--
👋 Hi!

🚨 Please read the following carefully before opening a new issue. Your issue may
be closed if it doesn't provide the required pieces of information. 🚨

We expect that it will take you about 30 minutes to produce a high-quality bug report.
While this may seem like a lot, putting care into issues helps us fix them faster.
For bug reports, it is REQUIRED to fill the rest of this template, or the issue will be closed.

Before submitting a new issue, please:

- Test using the latest release of the library, as maybe your bug has been already fixed.
- Check the existing [Issues](https://github.com/react-community/react-native-maps/issues?utf8=%E2%9C%93&q=) or on StackOverflow if there isn't a fix already for your problem.

Make sure you have followed the [Installation Instructions](https://github.com/react-community/react-native-maps/blob/master/docs/installation.md) too.

Still ready? Fill the template. 👇
-->

## Summary

While upgrading our app from Expo SDK 42 to 43 following the instructions from the official [upgrade blog post](https://blog.expo.dev/expo-sdk-43-aa9b3c7d5541), I ran into an issue where `npm i` couldn't proceed due to incompatible requirements for react-native-web. Expo 43 requires `0.17.1`, while react-native-maps is currently pinned to `^0.13.0`. 

After running `npm install --force` I can run our app just fine, so maybe this specific version pinning is not actually needed. Or just require a peer dependency with a version range instead of one very specific version.

## Environment info

<!--
  Please specify the react, react-native, react-native-maps versions, eg:

      react-native: 0.51.0
      react: 16.0.0
      react-native-maps: 0.19.0 or "git+ssh://git@github.com/react-native-maps/react-native-maps.git"

    ***Make sure you are on v0.48.0 or greater of react-native, otherwise you may get this error:

    Target Platform: eg: iOS - 11.2 Simulator/iPhone 6, Android 7.1.1/Galaxy S6/Genymotion etc

    ***If your target is iOS , please specify if you are using AppleMaps or GoogleMaps
-->

`npx react-native info` output:
```
info Fetching system and libraries information...
System:
    OS: Windows 10 10.0.19043
    CPU: (12) x64 Intel(R) Core(TM) i7-10750H CPU @ 2.60GHz
    Memory: 15.95 GB / 31.75 GB
  Binaries:
    Node: 16.12.0 - C:\ProgramData\scoop\apps\nvm\current\nodejs\nodejs\node.EXE
    Yarn: Not Found
    npm: 8.1.0 - C:\ProgramData\scoop\apps\nvm\current\nodejs\nodejs\npm.CMD
    Watchman: Not Found
  SDKs:
    Android SDK: Not Found
    Windows SDK: Not Found
  IDEs:
    Android Studio: Not Found
    Visual Studio: 16.11.31702.278 (Visual Studio Community 2019)
  Languages:
    Java: Not Found
  npmPackages:
    @react-native-community/cli: Not Found
    react: 17.0.1 => 17.0.1
    react-native: 0.64.2 => 0.64.2
    react-native-windows: ^0.64.0 => 0.64.25
  npmGlobalPackages:
    *react-native*: Not Found
```

Output of `npm ls --depth=0` (after running `npm install --force`, before that there is no valid install tree):
```
<our app>@ <path>
+-- @babel/core@7.15.8
+-- @expo/vector-icons@12.0.5
+-- @react-native-async-storage/async-storage@1.15.9
+-- @react-native-community/datetimepicker@3.5.2
+-- @react-native-community/slider@4.1.7
+-- @react-native-masked-view/masked-view@0.2.5
+-- @react-navigation/bottom-tabs@5.11.15
+-- @react-navigation/native@5.9.8
+-- @react-navigation/stack@5.14.9
+-- @types/react-native@0.64.18
+-- babel-preset-expo@8.5.1
+-- expo-auth-session@3.4.2
+-- expo-constants@12.1.3
+-- expo-crypto@10.0.3
+-- expo-font@10.0.3
+-- expo-haptics@11.0.3
+-- expo-image-picker@11.0.3
+-- expo-linear-gradient@10.0.3
+-- expo-location@13.0.4
+-- expo-notifications@0.13.3
+-- expo-permissions@13.0.3
+-- expo-random@12.0.1
+-- expo-splash-screen@0.13.4
+-- expo-web-browser@10.0.3
+-- expo@43.0.1
+-- jest-expo@43.0.1
+-- moment@2.29.1
+-- prettier@2.4.1
+-- react-dom@17.0.1
+-- react-native-gesture-handler@1.10.3
+-- react-native-map-link@2.8.1
+-- react-native-maps@0.28.0
+-- react-native-markdown-package@1.8.1
+-- react-native-modal@12.0.0
+-- react-native-render-html@5.1.1
+-- react-native-root-modal@5.0.1
+-- react-native-safe-area-context@3.3.2
+-- react-native-screens@3.8.0
+-- react-native-web@0.17.1
+-- react-native-windows@0.64.25
+-- react-native@0.64.2
+-- react@17.0.1
`-- typescript@4.3.5
```

## Steps to reproduce

<!--
- You must provide a clear list of steps and code to reproduce the problem.
- Keep the code reproducing the bug as simple as possible, with the minimum amount of code required to reproduce the issue. See https://stackoverflow.com/help/mcve.
- If this library has additional install steps, describe them (e.g., pod install? jetify? etc).
- Either re-create the bug using the repository's example app or link to a GitHub repository with code that reproduces the bug.
- Explain the steps we need to take to reproduce the issue:
-->

1. Take project that uses Expo SDK 42
2. Run `expo upgrade` to upgrade it to Expo SDK 43 (and follow other instructions from the [upgrade blog](https://blog.expo.dev/expo-sdk-43-aa9b3c7d5541)
3. Run `npm install`

Describe what you expected to happen:

<!--
  How did you expect your project to behave?
  It’s fine if you’re not sure your understanding is correct.
  Just write down what you thought would happen.
-->

`npm install` runs without errors

Describe what you actually happens:

<!--
  Did something go wrong?
  Is something broken, or not behaving as you expected?
  Describe this section in detail, and attach screenshots if possible.
  Don't just say "it doesn't work"!
-->

`npm install` produces an error:

```
$ npm i
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! While resolving: undefined@undefined
npm ERR! Found: react-native-web@0.17.1
npm ERR! node_modules/react-native-web
npm ERR!   react-native-web@"0.17.1" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer react-native-web@"^0.11" from react-native-maps@0.28.0
npm ERR! node_modules/react-native-maps
npm ERR!   react-native-maps@"0.28.0" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR!
npm ERR! See C:\Users\me\AppData\Local\npm-cache\eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\me\AppData\Local\npm-cache\_logs\2021-10-28T11_49_27_157Z-debug.log
```

_NB: I'm aware that Expo 43 doesn't use the very latest version of react-native-maps, but switching to 0.29.3 (current latest version) yields the exact same problem, except it wants 0.13 instead of 0.11_

## Reproducible sample code

<!--
  Please add minimal runnable repro as explained above so that the bug can be tested in isolation.
  If needed, you can also provide other samples: error messages / stack traces, screenshots, gifs, etc.

  Here are two ways to do it:

    * Create a new app using https://snack.expo.io/ and try to reproduce the issue in it.
      This is useful if you roughly know where the problem is, or can’t share the real code.

    * Or, copy your app and remove things until you’re left with the minimal reproducible demo.
      This is useful for finding the root cause. You may then optionally create a Snack.

  Once you’re done, copy and paste the link to the Snack or a public GitHub repository below:
-->

```sh
npm i -g expo-cli
mkdir expo-test
expo init                       # and accept all defaults...
cd my-app
expo install react-native-maps  # this errors out due to dependency conflicts,
                                # but it does add the package to package.json
npm install --force             # this resolved it for our situation, no guarantees though

Was this fix ever merged? I'm still having the same issue:

`npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: pins@0.1.0
npm ERR! Found: react-native-web@0.17.5
npm ERR! node_modules/react-native-web
npm ERR!   react-native-web@"^0.17.5" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer react-native-web@"^0.13" from react-native-maps@0.29.3
npm ERR! node_modules/react-native-maps
npm ERR!   react-native-maps@"*" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! `

--- a/package.json
+++ b/package.json
@@ -24,7 +24,7 @@
   "peerDependencies": {
     "react": ">= 16.0 || < 18.0",
     "react-native": ">= 0.51",
-    "react-native-web": "^0.13"
+    "react-native-web": ">= 0.11"
   },
   "devDependencies": {
     "@react-native-community/eslint-config": "3.0.1",

